src/library/scala/concurrent/Future.scala

```diff
@@ -604,22 +654,42 @@ object Future {
   *  @param executor  the execution context on which the future is run
   *  @return          the `Future` holding the result of the computation
   */
-  def apply[T](body: =>T)(implicit executor: ExecutionContext): Future[T] =
+  final def apply[T](body: =>T)(implicit executor: ExecutionContext): Future[T] =
     unit.map(_ => body)
 
-  /** Simple version of `Future.traverse`. Asynchronously and non-blockingly transforms a `IterableOnce[Future[A]]`
+  /** Starts an asynchronous computation and returns a `Future` instance with the result of that computation once it completes.
+  *
+  *  The following expressions are semantically equivalent:
+  *
+  *  {{{
+  *  val f1 = Future(expr).flatten
+  *  val f2 = Future.defer(expr)
+  *  val f3 = Future.unit.flatMap(_ => expr)
+  *  }}}
+  *
+  *  The result becomes available once the resulting Future of the asynchronous computation is completed.
+  *
+  *  @tparam T        the type of the result
+  *  @param body      the asynchronous computation, returning a Future
+  *  @param executor  the execution context on which the `body` is evaluated in
+  *  @return          the `Future` holding the result of the computation
+  */
+  final def defer[T](body: => Future[T])(implicit executor: ExecutionContext): Future[T] =
+    unit.flatMap(_ => body)
+
+  /** Simple version of `Future.traverse`. Asynchronously and non-blockingly transforms, in essence, a `IterableOnce[Future[A]]`
```

Comment:
Why "in essence"?

Also, should be "an \`IterableOnce[Future[A]]\`", I think

Reply:
I needed a way of saying "a subtype of ItrableOnce to the same subtype of IterableOnce". Suggest alternative?

Reply:
I have been thinking about this, and have not yet come up with an alternative (other than saying exactly what you said: "a subtype of ..."). I will continue to think on it

Reply:
@NthPortal Any suggestion? :)

Reply:
This isn't really a new line. That said, personally I would go for something less technical, like "transforms a collection of futures into a single future of a collection of the results of all the original futures (using the same collection type)"

Reply:
Fine by me